backend/dataall/modules/metadata_forms/services/metadata_form_service.py

```diff
@@ -295,25 +298,70 @@ def get_mf_permissions(entityUri):
     @MetadataFormAccessService.can_perform(UPDATE_METADATA_FORM_FIELD)
     def create_metadata_form_version(uri, copyVersion):
         with get_context().db_engine.scoped_session() as session:
+            mf = MetadataFormService.get_metadata_form_by_uri(uri)
             new_version = MetadataFormRepository.create_metadata_form_version_next(session, uri)
             if copyVersion:
                 mf_fields = MetadataFormRepository.get_metadata_form_fields(session, uri, copyVersion)
                 for field in mf_fields:
                     new_field = MetadataFormRepository.create_metadata_form_field(
                         session, uri, field.__dict__, new_version.version
                     )
+
+            all_attached = MetadataFormRepository.get_all_attached_metadata_forms(session, uri)
+            for attached in all_attached:
+                owner = MetadataFormService.get_entity_owner(attached)
+                if owner:
+                    NotificationRepository.create_notification(
+                        session,
+                        recipient=owner,
+                        target_uri=f'{attached.entityUri}|{attached.entityType}',
+                        message=f'New version {new_version.version} is available for metadata form "{mf.name}"',
+                        notification_type='METADATA_FORM_UPDATE',
```

Comment:
I think you need to update the enum dtype used in notification table. I am getting this error when trying to create a new MF version - and thus new notification   `ERROR: invalid input value for enum notificationtype: "METADATA_FORM_UPDATE"`

Reply:
Very strange, because I don't have this error and I haven't update any enums...
I ll look into it

Reply:
There is a migration `modify_notifications_column_types`,  that sets the type to simple Varchar
```
def upgrade():
    """
    Define column "type" as: type = Column(String, nullable=True)
    Rename column " " to " " both of type String
    """
    envname = os.getenv('envname', 'local')
    op.execute(f'ALTER TABLE {envname}.notification ALTER COLUMN "type" TYPE VARCHAR(100);')  # nosemgrep
    # semgrep finding ignored as no upstream user input is passed to the statement function
    # Only code admins will have access to the envname parameter of the f-string

    op.alter_column('notification', 'username', new_column_name='recipient', nullable=False, existing_type=sa.String())
    print('Notification columns updated')

```

Reply:
hmmm - guess my db somehow got out of sync. Re-running that migration script did change type of my notification.type column

will re test but this should be good as is then